src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/CommunityAPIReferralServiceTest.kt

```diff
@@ -15,41 +16,114 @@ class CommunityAPIReferralServiceTest {
 
   private val communityAPIClient = mock<CommunityAPIClient>()
 
+  private val sentAtDefault = OffsetDateTime.of(2020, 1, 1, 1, 1, 1, 1, ZoneOffset.UTC)
+  private val concludedAtDefault = OffsetDateTime.of(2020, 2, 2, 2, 2, 2, 2, ZoneOffset.UTC)
+
+  val communityAPIService = CommunityAPIReferralEventService(
+    "http://testUrl",
+    "/referral/sent/{id}",
+    "/referral/progress/{id}",
+    "/referral/end-of-service-report/{id}",
+    "secure/offenders/crn/{crn}/referral/start/context/{contextName}",
+    "secure/offenders/crn/{crn}/referral/end/context/{contextName}",
+    "commissioned-rehabilitation-services",
+    communityAPIClient
+  )
+
   @Test
-  fun `got service successfully`() {
-
-    val communityAPIService = CommunityAPIReferralEventService(
-      "http://testUrl",
-      "secure/offenders/crn/{id}/referral/sent",
-      "secure/offenders/crn/{crn}/referral/start/context/{contextName}",
-      "commissioned-rehabilitation-services",
-      communityAPIClient
-    )
+  fun `notify sent referral`() {
 
-    communityAPIService.onApplicationEvent(referralSentEvent)
+    communityAPIService.onApplicationEvent(getEvent(ReferralEventType.SENT))
 
     verify(communityAPIClient).makeAsyncPostRequest(
       "secure/offenders/crn/X123456/referral/start/context/commissioned-rehabilitation-services",
       ReferRequest(
         "ACC",
-        OffsetDateTime.of(2020, 1, 1, 1, 1, 1, 1, ZoneOffset.UTC),
+        sentAtDefault,
         123456789,
-        "http://testUrl/secure/offenders/crn/68df9f6c-3fcb-4ec6-8fcf-96551cd9b080/referral/sent",
+        "http://testUrl/referral/sent/68df9f6c-3fcb-4ec6-8fcf-96551cd9b080",
       )
     )
   }
 
-  private val referralSentEvent = ReferralEvent(
-    "source",
-    ReferralEventType.SENT,
-    SampleData.sampleReferral(
-      id = UUID.fromString("68df9f6c-3fcb-4ec6-8fcf-96551cd9b080"),
-      referenceNumber = "HAS71263",
-      crn = "X123456",
-      relevantSentenceId = 123456789,
-      serviceProviderName = "Harmony Living",
-      sentAt = OffsetDateTime.of(2020, 1, 1, 1, 1, 1, 1, ZoneOffset.UTC)
-    ),
-    "http://localhost:8080/sent-referral/68df9f6c-3fcb-4ec6-8fcf-96551cd9b080"
-  )
+  @Test
+  fun `notify cancelled referral`() {
+
+    communityAPIService.onApplicationEvent(getEvent(ReferralEventType.CANCELLED, concludedAtDefault))
+
+    verify(communityAPIClient).makeAsyncPostRequest(
+      "secure/offenders/crn/X123456/referral/end/context/commissioned-rehabilitation-services",
+      ReferralEndRequest(
+        "ACC",
+        sentAtDefault,
+        concludedAtDefault,
+        123456789,
+        "CANCELLED",
+        "http://testUrl/referral/progress/68df9f6c-3fcb-4ec6-8fcf-96551cd9b080",
```

Comment:
where do these urls come from? they don't seem right?

Reply:
These are purely fictitous for the purpose of the tests.